Implement Updater

SparklingComet opened the pull request:
Checks for updates upon enabling the plugin.

Todo:
- [x] Config setting to disable the updater.
- [x] Testing

SparklingComet committed:
Create Updater.java

Clone gist: https://gist.github.com/SparklingComet/673a4ef0b3896e5acb4af9c8b1da9d97

SparklingComet committed:
Update api url

SparklingComet committed:
Fix package declaration

SparklingComet committed:
Fix buildtype support

SparklingComet committed:
Implement Updater in main class

* Creates a new thread for async version checking.

SparklingComet committed:
Fix constructor call

SparklingComet committed:
Major fixes to the Updater (build type related)

NB: needs testing.

SparklingComet committed:
Fix field indentation

SparklingComet committed:
Add setting to disable updater

KillerOfPie commented:
Looks good to me, I'll probably have time to test it on Thursday if you don't by then.

SparklingComet commented:
I won't be able to test it as I'll still be out not town.
I have typed all of this on my tablet so there could be missing imports etc.

KillerOfPie committed:
Fixes
- Fixed issue with URL not catching exception
- Fixed issue with log never being set

- Tested & working

KillerOfPie commented:
@SparklingComet should be good now only had 2 issues

KillerOfPie committed:
Fixes
- Fixed version (1.6.0-Beta)

SparklingComet commented on pom.xml:
```diff
@@ -6,7 +6,7 @@
     <groupId>org.shanerx</groupId>
     <artifactId>tradeshop</artifactId>
-    <version>1.6.0-BETA</version>
+    <version>1.6.0-Updater</version>
```
What's this? This isn't a valid release tag. I assume you wanted to check wether the latest build had been compiled...?
If yes, this should be changed back to BETA.

KillerOfPie replied:
Correct, and I edited and pushed right before I left my house should already be fixed

SparklingComet commented on src/main/java/org/shanerx/tradeshop/Updater.java:
```diff
@@ -62,11 +64,19 @@
+		try {
+			url = new URL("https://api.spigotmc.org/legacy/update.php?resource=32762"); // Edit API URL.
+		} catch (MalformedURLException ex) {
+			log.log(Level.WARNING, "Error: Bad URL while checking {0} !", pdf.getName());
```
Maybe use String.format(String, String)

Btw, I completely forgot about that try..catch block.

SparklingComet replied:
Also, I think we could also use an empty catch block as that URL is a constant and is definitely a valid URL.

KillerOfPie replied:
Should always have an error output just incase, it should never but that doesn't mean it won't and that is the proper way for the url to be done.

SparklingComet commented on src/main/java/org/shanerx/tradeshop/Utils.java:
```diff
@@ -21,7 +21,11 @@
+import org.bukkit.ChatColor;
+import org.bukkit.Material;
+import org.bukkit.Nameable;
+import org.bukkit.OfflinePlayer;
```
Good one, thought we had ironed out all wildcard imports.

KillerOfPie replied:
I didn't even touch that, so my ide auto corrected that

SparklingComet commented on src/main/resources/plugin.yml:
```diff
@@ -1,5 +1,5 @@
 name: TradeShop
-version: 1.6.0-BETA
+version: 1.6.0-Updater
```
Copying from above:

> What's this? This isn't a valid release tag. I assume you wanted to check wether the latest build had been compiled...?
If yes, this should be changed back to BETA.

KillerOfPie replied:
Same as above

SparklingComet commented:
If you have tested it, it should be ready for merge.

KillerOfPie commented:
I've tested it to the point that it tells me im ahead when im working on things

SparklingComet commented:
I assume I will merge it with ``1.6`` then.
I will not delete the ``Updater`` branch yet, though.
